src/cache.c

```diff
@@ -9,19 +9,22 @@
  */
 struct cache_entry *alloc_entry(char *path, char *content_type, void *content, int content_length)
 {
-    ///////////////////
-    // IMPLEMENT ME! //
-    ///////////////////
+    struct cache_entry *e = malloc(sizeof(struct cache_entry));
+    e->path = strdup(path);
```

Comment:
Can you approach this differently? What is a struct? Can you do something like this? 
e->path=malloc(strlen(path)+1);
strcpy(e->path, path)